tycho-common/src/storage.rs

```diff
@@ -479,6 +481,44 @@ pub trait ProtocolGateway {
         pagination_params: Option<&PaginationParams>,
     ) -> Result<WithTotal<Vec<String>>, StorageError>;
 }
+pub struct EntryPointFilter {
+    pub protocol_system: Option<ProtocolSystem>,
+}
+
+impl EntryPointFilter {
+    pub fn new(protocol: Option<String>) -> Self {
+        Self { protocol_system: protocol }
+    }
+}
+
+// Trait for entry point gateway operations.
+#[async_trait]
+pub trait EntryPointGateway {
+    /// Upserts a list of entry points into the database.
+    async fn upsert_entry_points(
+        &self,
+        entry_points: &[EntryPointWithData],
+        component_id: &str,
+    ) -> Result<(), StorageError>;
+
+    /// Retrieves a list of entry points with their tracing data from the database.
+    async fn get_entry_points_with_data(
+        &self,
+        filter: EntryPointFilter,
+    ) -> Result<Vec<EntryPointWithData>, StorageError>;
+
+    /// Upserts a list of traced entry points into the database.
+    async fn upsert_traced_entry_points(
+        &self,
+        traced_entry_points: &[TracedEntryPoint],
+    ) -> Result<(), StorageError>;
+
+    /// Retrieves the tracing results for an entry point from the database.
+    async fn get_traced_entry_point(
```

Comment:
We also need:

async fn get_traced_entry_point_with_data (
    &self,
    entry_point: EntryPointWithData,
) -> Result<TracingResult>, StorageError>;